### PR TITLE
arduino: Update to version 2.0.0

### LINKED ISSUE #9267: Package Update Request
#### Scoop Configuration

The package Arduino is updated to 2.0.0 but scoop hasn't updated since version 1.8.19 i.e. 9 months ago
Official Site link - https://www.arduino.cc/en/software
Download link - https://downloads.arduino.cc/arduino-ide/arduino-ide_2.0.0_Windows_64bit.zip

### PR DESCRIPTION
**Pull request by sayeed205**
Updated version for Arduino IDE
Closes #9267 

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/arduino.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+    "version": "2.0.0",
     "description": "Arduino IDE",
     "homepage": "https://www.arduino.cc",
     "license": "GPL-2.0-or-later",
-    "url": "https://downloads.arduino.cc/arduino-1.8.19-windows.zip",
-    "hash": "sha512:c4072d808aea3848bceff5772f9d1e56a0fde02366b5aa523d10975c54eee2ca8def25ee466abbc88995aa323d475065ad8eb30bf35a2aaf07f9473f9168e2da",
-    "extract_dir": "arduino-1.8.19",
+    "url": "https://downloads.arduino.cc/arduino-ide/arduino-ide_2.0.0_Windows_64bit.zip",
+    "hash": "sha512:bd1b129030abc6185e69057cb3677448854e3cbdc5e984abcc093d3e1add84474794380ad03c417c20bafd265ba9f7567fde51874151f00af4593da231df6a81",
+    "extract_dir": "arduino-2.0.0",
```
Comment (tech189) [inline bucket/arduino.json:8]: Remove extract_dir as this directory no longer exists in version 2.0.0. Please also remove it from autoupdate.

File: bucket/arduino.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
         "regex": "Arduino IDE ([\\d.]+)<"
     },
     "autoupdate": {
-        "url": "https://downloads.arduino.cc/arduino-$version-windows.zip",
+        "url": "https://downloads.arduino.cc/arduino-ide/arduino-ide_$version_Windows_64bit.zip",
         "hash": {
             "url": "$baseurl/arduino-$version.sha512sum.txt"
         },
```
Comment (tech189) [inline bucket/arduino.json:40]:
```suggestion
```
Remove hash from autoupdate as hashes are no longer provided on the website. (Hashes on GitHub releases are only for the .exe download, sadly)

**Review by tech189 — CHANGES_REQUESTED** [review]
Thanks for your PR! I've made a few fixes - in future do test out your manifest with `scoop install <path to your manifest>` to check it installs.

**Comment by tech189** [comment]
I think we have to also remove the `pre_install` script, the `post_install` script, and the `persist` section as none of the files referred to exist in this version. I think the IDE now stores the plugins, preferences, and hardware config in `\AppData\Local\Arduino15`

**Comment by tech189** [comment]
> I think we have to also remove the `pre_install` script...

Thanks for applying my review suggestions! Could you please follow my suggestions I partially quoted above?

The extract_dir needs to be removed too as it currently fails to install.

**Comment by sayeed205** [comment]
> The extract_dir needs to be removed too as it currently fails to install.

You mean the [auto update extract_dir](https://github.com/sayeed205/Extras/blob/b3807eee5171395285cd1eb744a0a69a565ae02d/bucket/arduino.json#L38) or [installation extract_dir](https://github.com/sayeed205/Extras/blob/b3807eee5171395285cd1eb744a0a69a565ae02d/bucket/arduino.json#L8)

**Comment by tech189** [comment]
Both as the enclosing folder "arduino-..." is no longer in the zip package, all files are in the root folder.

**Comment by sayeed205** [comment]
> Both as the enclosing folder "arduino-..." is no longer in the zip package, all files are in the root folder.

Sir you do necessary changes. Actually I(noob) dunno much about how scoop does things.

**Review by tech189 — APPROVED** [review]

**Review by issaclin32 — APPROVED** [review]

**Pull request merged**